### PR TITLE
Validate the link on the image is valid

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Protect the user from entering garbage text into the link field.

# Needed By (Date)
- asap

# Urgency
- medium

# Steps to Test
1. checkout this branch
1. in a wysiwyg add an image
1. for the link url enter something like `this wont work`
1. ensure it gives you a form error.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
POKE

File: src/Plugin/MediaEmbedDialog/Image.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function isApplicable() {
    *
    * @see Linkit::processLinkitAutocomplete()
    */
-  public static function processLinkitAutocomplete(&$element, FormStateInterface $form_state, &$complete_form) {
```
Comment (sherakama) [inline src/Plugin/MediaEmbedDialog/Image.php]: I suspect that no one is using this static function outside of this module but it is something to think about when versioning.

File: src/Plugin/MediaEmbedDialog/Image.php [hunk 2]
```diff
@@ -411,7 +443,7 @@ public function preRender(array $element) {
    * @return \Drupal\Core\Url
    *   Constructed url object.
    */
-  protected function getLinkObject($link_path, array $link_options = []) {
+  protected static function getLinkObject($link_path, array $link_options = []) {
```
Comment (sherakama) [inline src/Plugin/MediaEmbedDialog/Image.php]: Does this need to be static?

Comment (pookmish) [reply src/Plugin/MediaEmbedDialog/Image.php]: it doesn't need to be static but it doesn't use `$this` which allows it to be static. :man_shrugging:

**Pull request merged**